### PR TITLE
(mostly) Use wxArtProvider for the UI icons instead of the XPM icons.

### PR DESCRIPTION
**Pull request by jcontoso**
This change is intended to make the toolbar and contents panel icons look native (especially under GTK environments).

### CODE REVIEW

**Comment by rzvncj** [comment]
Thanks for the pull request! For starters, please remove the "formatting fixes" patches - all you need to do is run `clang-format -i file.cpp` on the files you've modified (before committing), and they'll be correctly indented.

**Comment by jcontoso** [comment]
> Thanks for the pull request! For starters, please remove the "formatting fixes" patches - all you need to do is run `clang-format -i file.cpp` on the files you've modified (before committing), and they'll be correctly indented.

Thanks for letting me know.

**Comment by rzvncj** [comment]
> > Thanks for the pull request! For starters, please remove the "formatting fixes" patches - all you need to do is run `clang-format -i file.cpp` on the files you've modified (before committing), and they'll be correctly indented.
> 
> Thanks for letting me know.

No problem. The rules are in https://github.com/rzvncj/xCHM/blob/master/src/.clang-format, that's what `clang-format` will use.

**Comment by jcontoso** [comment]
It should be all fine now.

**Comment by rzvncj** [comment]
Sure, but would it be possible, instead of having the last three commits that just fix the formatting of the code to actually remove them completely, then amend the first commit to be properly formatted in the first place?

Something like this? https://www.git-tower.com/learn/git/faq/git-squash

**Comment by jcontoso** [comment]
Alright. I squashed the commits.

File: src/chmframe.cpp [hunk 1]
```diff
@@ -61,26 +62,44 @@
...
+    "(quotes not included) will find all the pages that contain both word1 and "
+    "word2, in whatever order.</li>"
+    "<li>Allowing partial matches will find pages that contain words that only "
+    "start with the typed strings, "
+    "i.e. searching for \'ans 4\' (quotes not included) will find pages that "
+    "contain the sentence "
+    "\'The answer is 42.\'.</li><li>Right clicking on the displayed page "
+    "brings out a popup menu with "
+    "common options.</li></ul><br><br>Ctrl-'c' is copy, Ctrl-'f' is find in "
+    "page, Ctrl-'=' is zoom-in, "
```
Comment (rzvncj) [inline src/chmframe.cpp]: We must have different `clang` versions, my `clang-format -i` session does not cause the above (pull-request unrelated) changes. FWIW, I'm using clang version 14.0.6, on Arch Linux.

Could this part be omitted from the patch? If not, I'll see about it after the pull request is merged.

File: src/chmframe.cpp [hunk 2]
```diff
@@ -688,11 +707,8 @@ wxMenuBar* CHMFrame::CreateMenu()
 }
 
 namespace {
-
 #include <hbook_closed.xpm>
 #include <hbook_open.xpm>
-#include <hpage.xpm>
-
 } // namespace
```
Comment (rzvncj) [inline src/chmframe.cpp:696]: Why remove the empty newlines here (and below)? I'd prefer that we keep the current style unless there's a good reason to change it (or something like `clang-format` does it, which is not the case here).

There are also other anonymous namespaces in other files (such as chmfile.cpp), and there's always a newline under `namespace {`. This change make the namespace coding style inconsistent.

File: src/chmframe.cpp [hunk 3]
```diff
@@ -839,44 +855,49 @@ void CHMFrame::SaveExitInfo()
...
-#include <fileopen.xpm>
-#include <find.xpm>
-#include <forward.xpm>
 #include <fullscreen.xpm>
-#include <helpicon.xpm>
-#include <home.xpm>
 #include <htmoptns.xpm>
 #include <htmsidep.xpm>
-#include <print.xpm>
-
```
Comment (rzvncj) [inline src/chmframe.cpp:855]: Please see previous comment on removing these newlines.

**Review by rzvncj — CHANGES_REQUESTED** [review]

File: src/chmframe.cpp (hunk 2 above)
Comment (jcontoso) [reply src/chmframe.cpp:696]: Fixed with the latest commit.

File: src/chmframe.cpp (hunk 3 above)
Comment (jcontoso) [reply src/chmframe.cpp:855]: Fixed with the latest commit.

File: src/chmframe.cpp (hunk 1 above)
Comment (jcontoso) [reply src/chmframe.cpp]: Manually fixed.

**Comment by rzvncj** [comment]
Thanks! I'll spare you the further squashing and merge.

**Pull request merged**